File: dsa/6.hash_table/771_Jewels_and_Stones.java
package hash_table;
import java.util.*;

class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        Set<Character> hashSet = new HashSet<Character>();
        int result = 0;

        for (int i = 0; i < jewels.length(); i++){
            if (hashSet.contains(jewels.charAt(i)) == false){
                hashSet.add(jewels.charAt(i));
            }
        }

        for (int j = 0; j < stones.length(); j++){
            if (hashSet.contains(stones.charAt(j)) == true){
                result += 1;
            }
        }
        return result;
    }
}

File: dsa/6.hash_table/771_Jewels_and_Stones.py
import typing as t

class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        jewels_set: t.Set[str] = set()
        result: int = 0
        for jew in jewels:
            if jew not in jewels_set:
                jewels_set.add(jew)
        
        for stone in stones:
            if stone in jewels_set:
                result += 1
        
        return result